src/fheroes2/dialog/dialog_selectcount.cpp

```diff
@@ -135,6 +136,11 @@ bool Dialog::SelectCount( std::string header, const int32_t min, const int32_t m
             redraw_count = true;
         }
 
+        if ( uiElement ) {
+            uiElement->processEvents( uiOffset );
+            display.render();
+        }
```

Comment:
When `uiElement` is not `nullptr` the `display.render()` method is called on every loop. It results in very high render calls:
![изображение](https://github.com/user-attachments/assets/70cfbdff-a0fa-417a-ae23-540556418d22)
We can make `processEvents()` to return `true` if `render()` is needed but to do this we will need to update `DialogElement` class and all its inheritances.

So to solve only this issue we can check for mouse action in the UI element area and only then process the events and do `render()`.
Like this (a code without any optimizations):
```suggestion
        if ( uiElement
             && ( le.isMouseLeftButtonReleasedInArea( { uiOffset, uiElement->area() } ) || le.isMouseRightButtonPressedInArea( { uiOffset, uiElement->area() } ) ) ) {
            uiElement->processEvents( uiOffset );
            display.render();
        }
```

Reply:
Done